### PR TITLE
Dev/testing

### LINKED ISSUE #21: Testing
As a developer, I can run `yarn test` which allows me to run my Testing Library Tests

We are not intended to enforce this but allowing those who would like to use it (e.g. TDD or bug isolation) to do so.

Additionally, this will 
1. Not block builds (no Github Action usage) and 
2. Not run on an Git hooks (e.g. using `Husky`)

### PR DESCRIPTION
**Pull request by johntimothybailey**
closes #21 

If you'd like to see more example testing or more unit level testing ("white box") with Enzyme, please let me know. In the meantime and in order to understand the existing components, I setup tests for four components.

### CODE REVIEW

File: KukaApp/.gitignore [hunk 1]
```diff
@@ -61,3 +61,7 @@ buck-out/
 
 # CocoaPods
 /ios/Pods/
+
+
+# Testing
+__tests__/.coverage/
```
Comment (johntimothybailey) [inline KukaApp/.gitignore:67]: As a developer I can now do `yarn test --coverage` (again this is just a optional preference that comes with Jest and REALLY helpful for developers like me)

File: KukaApp/package.json [hunk 2]
```diff
@@ -39,16 +39,15 @@
...
+    "@testing-library/react-native": "^7.2.0",
     "@ui-kitten/metro-config": "5.0.0",
     "babel-jest": "^24.9.0",
     "eslint": "^6.5.1",
     "jest": "^24.9.0",
+    "jest-svg-transformer": "^1.0.0",
     "metro-react-native-babel-preset": "^0.58.0",
     "prettier": "^2.0.2",
     "react-test-renderer": "16.11.0"
-  },
```
Comment (johntimothybailey) [inline KukaApp/package.json:49]: For uniformity and leveraging `$ npx jest --init`

File: KukaApp/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
+// For a detailed explanation regarding each configuration property, visit:
```
Comment (johntimothybailey) [inline KukaApp/jest.config.js:1]: Generated from `$ npx jest --init`

File: package-lock.json [hunk 4]
```diff
@@ -1,11 +0,0 @@
-{
```
Comment (johntimothybailey) [inline package-lock.json:1]: As indicated in the single commit, this is dead code and should be removed. It caused me confusion when initially setting up the project.

File: KukaApp/__tests__/App-test.tsx [hunk 5]
```diff
@@ -1,14 +0,0 @@
-/**
```
Comment (johntimothybailey) [inline KukaApp/__tests__/App-test.tsx:1]: Not being used. It appears to be boilerplate code

File: KukaApp/__tests__/BootComponent.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+export const BootComponent = props => {
+  return (
+    <>
+      <IconRegistry icons={EvaIconsPack} />
+      <ApplicationProvider
+        {...eva}
+        theme={{ ...eva.light, ...theme }}
+        customMapping={mapping}
+      >
+        <SafeAreaProvider
```
Comment (johntimothybailey) [inline KukaApp/__tests__/BootComponent.js:18]: Components are referencing the "safe area" as such we'll need to make this a standard. This benefits us too if we use this for testing the screens. Granted I usually do a `BootScreen` and `BootComponent` as those each should have a good amount of separation of concerns.

**Review by johnkmzhou — APPROVED** [review]
lgtm

**Pull request merged**